File: 217. Contains Duplicate/Solution.cs
public class Solution {
    public bool ContainsDuplicate(int[] nums) {
        ISet<int> numset = new HashSet<int>();

        foreach (int num in nums) {
            if (numset.Contains(num)) {
                return true;
            }
            numset.Add(num);
        }
        return false;
    }
}

// Time complexity: O(n)
// Space complexity: O(n)

File: 217. Contains Duplicate/Solution.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> numset = new HashSet<>();

        for (int i = 0; i < nums.length; i++) {
            if (numset.contains(nums[i])) {
                return true;
            }
            numset.add(nums[i]);
        }
        return false;
    }
}

// Time complexity: O(n)
// Space complexity: O(n)